modules/flann/include/opencv2/flann/lsh_table.h

```diff
@@ -245,7 +245,7 @@ class LshTable
     {
         std::cerr << "LSH is not implemented for that type" << std::endl;
         assert(0);
-        return 1;
+        return static_cast<size_t>(-1);
```

Comment:
I'm not sure that this is reliable for 32bit OS - the values are different

Reply:
Errors return -1 and not 1 in FLANN. As the return type is an unsigned integer, value needed to be casted.
However, as an error value for a key, 0 should do the work to help the user to figure out what is going on.